Utilities/XrdAdaptor/src/XrdSource.cc

```diff
@@ -103,6 +105,11 @@ Source::Source(timespec now, std::unique_ptr<XrdCl::File> fh, const std::string
     setXrootdSite();
     assert(m_qm.get());
     assert(m_fh.get());
+    XrdSiteStatisticsInformation *statsService = XrdSiteStatisticsInformation::getInstance();
+    if (statsService)
+    {
+        m_stats = statsService->getStatisticsForSite(m_site);
```

Comment:
It looks to me like ```getStatisticsForSite``` is not thread safe. Are ```Source```s ever created on different threads?